testing-basics-garden-project/garden-test015.js

```diff
@@ -0,0 +1,37 @@
+import { Garden } from "../src/Garden";
+import { FruitTree } from "../src/FruitTree";
+const chai = require("chai");
+const assert = chai.assert;
+
+describe("Garden tests", function () {
+   let myGarden = new Garden();
+   let myFruitTree = new FruitTree("Strawberry");
+   it("should be able to create a Garden", function () {
+      assert.instanceOf(myGarden, Garden);
+   });
+
+   it("should be able to add fruit trees", function () {
+      myGarden.addPlant(myFruitTree);
+      assert.isAbove(myGarden.fruitTrees.length, 0);
+   })
+
+   it("should not allow more than 5 fruit trees", function () {
+      for (let i = 0; i < 6; i++) {
+         myGarden.addPlant(myFruitTree);
+      }
+      assert.isAtMost(myGarden.fruitTrees.length, 5);
+   });
+});
+
+/* Our tests pass, but they are not independent of other tests
+   This can cause problems for future tests
+   We can setup what happens before each test using "beforeEach"
+   Use "beforeEach" to setup the myGarden variable for every test case
+   e.g.
+      describe("Tests", function(){
+         let initialVar
+         beforeEach(() => {
+            // setup code for initialVar
+         })
+      })
+*/
```

Comment:
Tried to clarify things a bit and change the example to something that's closer to the final code. Hopefully it's not just giving the answer away. What do you think?

````
/* 
  Our tests pass, but they are not independent of other tests.
  This can cause problems for future tests.
  We can set up what happens before each test using the `beforeEach` hook.
  Just below your `myGarden` variable, use `beforeEach` to set up the `myGarden` variable for every test case, for example:
  ```
  describe("Tests", function(){
    let Plane
    beforeEach(() => {
      Plane = new Plane()
    })
  })
  ```
*/
````

Reply:
Hi Kris, 

This is good! It's more explicit than what I originally have.
I don't think it gives the answer away.